tool/ci/pkg/environment/environment_test.go

```diff
@@ -0,0 +1,243 @@
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewEnvironment(t *testing.T) {
+	pr := &PullRequestMetadata{Author: "Codertocat",
+		RepoName:   "Hello-World",
+		RepoOwner:  "Codertocat",
+		Number:     2,
+		HeadSHA:    "ec26c3e57ca3a959ca5aad62de7213c562f8c821",
+		BaseSHA:    "f95f852bd8fca8fcc58a9a2d6c842781e32a215e",
+		BranchName: "changes",
+	}
+	tests := []struct {
+		cfg        Config
+		checkErr   require.ErrorAssertionFunc
+		expected   *Environment
+		desc       string
+		createFile bool
+	}{
+		{
+			cfg: Config{
+				Client:        github.NewClient(nil),
+				EventPath:     "",
+				unmarshalRevs: UnmarshalReviewersTest,
+				Token:         "testtoken",
+			},
+			checkErr:   require.Error,
+			desc:       "invalid Environment config with Reviewers parameter",
+			expected:   nil,
+			createFile: true,
+		},
+		{
+			cfg: Config{
+				Client:        github.NewClient(nil),
+				Reviewers:     `{"foo": ["bar", "baz"], "":["admin"]}`,
+				unmarshalRevs: UnmarshalReviewersTest,
+				Token:         "testtoken",
+			},
+			checkErr: require.NoError,
+			desc:     "valid Environment config",
+			expected: &Environment{
+				reviewers:        map[string][]string{"foo": {"bar", "baz"}, "": {"admin"}},
+				Client:           github.NewClient(nil),
+				PullRequest:      pr,
+				defaultReviewers: []string{"admin"},
+			},
+			createFile: true,
+		},
+		{
+			cfg: Config{
+				Client:        github.NewClient(nil),
+				Reviewers:     `{"foo": ["bar", "baz"], "":["admin"]}`,
+				unmarshalRevs: UnmarshalReviewersTest,
+				Token:         "testtoken",
+			},
+			checkErr: require.NoError,
+			desc:     "valid Environment config",
+			expected: &Environment{
+				reviewers:        map[string][]string{"foo": {"bar", "baz"}, "": {"admin"}},
+				Client:           github.NewClient(nil),
+				PullRequest:      pr,
+				defaultReviewers: []string{"admin"},
+			},
+			createFile: true,
+		},
+		{
+			cfg: Config{
+				Client:        github.NewClient(nil),
+				Reviewers:     `{"foo": ["bar", "baz"]}`,
+				unmarshalRevs: UnmarshalReviewersTest,
+				Token:         "testtoken",
+			},
+			checkErr:   require.Error,
+			desc:       "invalid Environment config, has no default reviewers key",
+			expected:   nil,
+			createFile: true,
+		},
+		{
+			cfg: Config{
+				Reviewers:     `{"foo": "baz", "":["admin"]}`,
+				Client:        github.NewClient(nil),
+				unmarshalRevs: UnmarshalReviewersTest,
+				Token:         "testtoken",
+			},
+			checkErr:   require.Error,
+			desc:       "invalid reviewers object format",
+			expected:   nil,
+			createFile: true,
+		},
+		{
+			cfg: Config{
+				unmarshalRevs: UnmarshalReviewersTest,
+				Token:         "testtoken",
+			},
+			checkErr:   require.Error,
+			desc:       "invalid config with no client",
+			expected:   nil,
+			createFile: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			if test.createFile {
+				f, err := ioutil.TempFile("", "payload")
```

Comment:
Use `t.TempFile()` instead and there will be no need to remove it.

Reply:
I'm not seeing a `t.TempFile()` function

Reply:
My mistake - there's only `t.TempDir()`. This is fine as-is